hasher-matcher-actioner/webapp/src/pages/MatchIndex.jsx

```diff
@@ -0,0 +1,251 @@
+          />
+          <InputGroup.Append>
+            <Button onClick={loadSearchPage}>Search</Button>
+          </InputGroup.Append>
+        </InputGroup>
+        <Form.Text>
+          Search for matches of your content IDs or Signal IDs.
+        </Form.Text>
+      </Form.Group>
+    </Form>
+  );
+}
+
+MatchListFilters.propTypes = {
+  searchInAttribute: PropTypes.oneOf(['contentId', 'signalId']),
+  searchQuery: PropTypes.string,
+};
+
+MatchListFilters.defaultProps = {
+  searchInAttribute: 'contentId',
+  searchQuery: undefined,
+};
+
+function MatchList({selection, onSelect, searchInAttribute, searchQuery}) {
+  const [matchesData, setMatchesData] = useState(null);
+
+  useEffect(() => {
+    let apiPromise;
+    if (searchInAttribute === 'contentId') {
+      apiPromise = fetchMatchesFromContent(searchQuery);
+    } else if (searchInAttribute === 'signalId') {
+      /** Hack alert. This expects strings like
+       * "facebook/threatexchange|12312121" to be the searchQuery. A pipe
+       * separates the source from the signal id */
+      const parts = searchQuery.split('|');
+      apiPromise = fetchMatchesFromSignal(parts[0], parts[1]);
+    } else {
+      apiPromise = fetchAllMatches();
+    }
+
+    apiPromise.then(matches => setMatchesData(matches.match_summaries));
+    // .catch(err => console.log(err));
+  }, [searchInAttribute, searchQuery]);
+
+  return (
+    <>
+      <Spinner hidden={matchesData !== null} animation="border" role="status">
+        <span className="sr-only">Loading...</span>
+      </Spinner>
+      <Collapse in={matchesData !== null}>
+        <div className="row mt-3">
+          <div className="col-12">
+            <table className="table table-hover small">
+              <thead>
+                <tr>
+                  <th>Content Id</th>
+                  <th>Matched in Dataset</th>
+                  <th>Seen</th>
+                </tr>
+              </thead>
+              <tbody>
+                {matchesData !== null && matchesData.length ? (
+                  matchesData.map(match => (
+                    <tr
+                      className={classNames('align-middle', {
+                        'table-info':
+                          match.content_id === selection.contentId &&
+                          match.signal_id === selection.signalId,
+                      })}
+                      onClick={e =>
+                        onSelect({
+                          contentId: match.content_id,
+                          signalId: match.signal_id,
+                          signalSource: match.signal_source,
+                          [e.target.name]: e.target.value,
+                        })
+                      }
+                      key={`${match.signal_source}_${match.signal_id}_${match.content_id}`}>
+                      <td className="align-middle">{match.content_id}</td>
+                      <td className="align-middle">{match.signal_source}</td>
```

Comment:
I think we've talked about this and know we are currently refining vernacular but signal_source != dataset

Can you add a note to change later if you don't want to scope that change of querying the other db object to get datasets in this diff? Thanks!